ray-curator/ray_curator/stages/classifiers/aegis.py

```diff
@@ -0,0 +1,555 @@
+
+        if self.add_instruction_data_guard:
+            df_cpu[self.prob_column] = collected_output["preds"].tolist()
+            df_cpu[self.pred_column] = (collected_output["preds"] >= 0.5).tolist()  # noqa: PLR2004
+        else:
+            df_cpu[self.pred_column] = collected_output["preds"].tolist()
+
+        return df_cpu
+
+    def process(self, batch: DocumentBatch) -> DocumentBatch:
+        processed_outputs = []
+        df_cpu = batch.to_pandas()
+
+        for model_input_batch in self.yield_next_batch(df_cpu):
+            # Forward pass
+            with torch.no_grad():
+                outputs = self.model(model_input_batch)
+
+            processed_output = self.process_model_output(outputs)
+            processed_outputs.append(processed_output)
+
+        # Collect all outputs
+        collected_output = self.collect_outputs(processed_outputs)
+
+        # Create output Pandas DataFrame
+        df_cpu = self.create_output_dataframe(df_cpu, collected_output)
+
+        # Sort by seq_order to preserve original order from tokenizer
+        if self.has_seq_order:
+            df_cpu = df_cpu.sort_values(by="_curator_seq_order", ignore_index=True).drop(
+                columns=["_curator_seq_order"]
+            )
+
+        return DocumentBatch(
+            task_id=batch.task_id,
+            dataset_name=batch.dataset_name,
+            data=df_cpu,
+            _metadata=batch._metadata,
+            _stage_perf=batch._stage_perf,
+        )
+
+
+@dataclass
+class WrapInPromptStage(ProcessingStage[DocumentBatch, DocumentBatch]):
+    """
+    WrapInPromptStage is a stage that truncates and wraps the input text in a prompt for the AEGIS model.
+    """
+
+    text_field: str
+    max_chars: int
+    _name = "wrap_in_prompt"
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.text_field]
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], ["_hidden_text"]
+
+    def _wrap_in_prompt(self, df: pd.DataFrame) -> pd.DataFrame:
+        documents = df[self.text_field].tolist()
+        prompts = [format_aegis(doc[: self.max_chars]) for doc in documents]
+        df["_hidden_text"] = prompts
+        return df
+
+    def process(self, batch: DocumentBatch) -> DocumentBatch:
+        df = batch.to_pandas()
+        df = self._wrap_in_prompt(df)
+
+        return DocumentBatch(
+            task_id=batch.task_id,
+            dataset_name=batch.dataset_name,
+            data=df,
+            _metadata=batch._metadata,
+            _stage_perf=batch._stage_perf,
+        )
+
+
+@dataclass
+class PostProcessResponsesStage(ProcessingStage[DocumentBatch, DocumentBatch]):
+    """
+    PostProcessResponsesStage is a stage that post-processes the responses from the AEGIS model.
+    """
+
+    pred_column: str
+    raw_pred_column: str
+    keep_raw_pred: bool
+    _name = "postprocess_responses"
+
+    def inputs(self) -> tuple[list[str], list[str]]:
```

Comment:
I believe this should be an ActorStage ie. in RayData can you tag this as `IS_ACTOR_STAGE` given that it has a setup.. and similarly should the `setup_on_node` to ensure the tokenizers are downloaded before we start a stage.


Also I saw  this https://github.com/nvidia-cosmos/cosmos-xenna/blob/c62aa91db62d14dcbf20cba69cfe3247b95016a9/examples/simple_vlm_inference.py#L129-L155 

What I really liked was `local_files_only=True` during setup, this guarantees that we don't waste time during "processing"